src/components/RightHandMenu/DonutChart/Donut.js

```diff
@@ -107,46 +211,61 @@ const renderActiveShape = (props) => {
  * for Race and similar type data
  * data slice filtered as per radioSelct and ToggleSelect to be used in place of mockData
  */
-function Donut() {
+function Donut(props) {
   const [legend, setLegend] = useState([])
   const [sum, setSum] = useState(0)
 
+  const [activeLabel, setActiveLabel] = useState(0)
+
+  const onPieEnter = useCallback(
+    (_, idx) => {
+      setActiveLabel(idx)
+    }, [setActiveLabel]
+  )
+
+  const { data, dataType } = props
+
   useEffect(() => {
     const l = []
     let sum1 = data.reduce(function (a, b) {
       return a + b.value
     }, 0)
-    data.map((entry, index) => (l.push({ key : entry.key, color : COLORS[index % COLORS.length], value : entry.value  })))
+    data.map((entry, index) => (l.push({ key : entry.key, color : COLORS[index % COLORS.length], value : entry.value, percent: entry.percent !== undefined ? entry.percent : null  })))
```

Comment:
Could you make this more reader friendly?  Also, did you choose `l` as a variable name for a specific reason?  Maybe something like this would be easier on the eyes... it's still a lot to unwrap, so maybe create a separate variable for the the `l.push...`  
```
data.map( (entry, index) => (l.push({ 
  key: entry.key, 
  color: COLORS[index % COLORS.length], 
  value: entry.value, 
  percent: entry.percent !== undefined ? entry.percent : null  
})))
```